### PR TITLE
1.2

### PR DESCRIPTION
**Pull request by timsuchanek**

### CODE REVIEW

File: src/commands/init.ts [hunk 1]
```diff
@@ -14,8 +14,11 @@ import {
   couldNotCreateProjectMessage,
   projectAlreadyExistsMessage,
   projectFileSuffix,
-  sampleSchemaURL, invalidSchemaFileMessage
+  sampleSchemaURL,
+  invalidSchemaFileMessage,
+  invalidProjectNameMessage, cantCopyAcrossRegions,
 } from '../utils/constants'
+import { isValidProjectName } from '../utils/validation'
```
Comment (schickling) [inline src/commands/init.ts]: Multiline import should always be after single line imports

File: src/index.ts [hunk 2]
```diff
@@ -85,7 +85,8 @@ async function main() {
       const name = argv['name'] || argv['n']
       const alias = argv['alias'] || argv['a']
       const region = argv['region'] || argv['r']
-      const schemaUrl = argv['schema'] || argv['s']
+      // if there are exactly 2 argv's, use the 2nd argument as an alternative for the schemaUrl
```
Comment (schickling) [inline src/index.ts]: We should find an abstraction where we can test CLI arguments. This seems like a case that could easily fail.

File: src/utils/constants.ts [hunk 3]
```diff
@@ -143,9 +143,17 @@ Whoops, something went wrong while creating the project.
...
+Found ${projectFiles.length} project ${projectFiles.length === 1 ? 'file' : 'files'} in the current directory. Looks like you've already setup your backend.
+You can still use graphcool init --output file.graphcool to specify a custom output file.
 `
 
+export const cantCopyAcrossRegions = `\
+A project can't be copied across regions. Please specify the --copy parameter without --region.
+`
+
+export const invalidProjectNameMessage = (projectName: string) => `\
+'${projectName}' is not a valid project name. It must begin with an uppercase letter.`
```
Comment (schickling) [inline src/utils/constants.ts]: Use new lines at the end in a consistent way.

File: src/utils/ping.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import { Region } from '../types'
+import * as cuid from 'cuid'
+import * as bluebird from 'bluebird'
+import {sum} from 'lodash'
+
+async function getPing(url: string) {
```
Comment (schickling) [inline src/utils/ping.ts]: I'd call this `runPing` + it's missing the return type

File: tests/init.test.ts [hunk 5]
```diff
@@ -13,6 +13,7 @@ import { simpleTwitterSchema } from './fixtures/schemas'
 import 'isomorphic-fetch'
 import { readProjectIdFromProjectFile, readVersionFromProjectFile } from '../src/utils/file'
 import TestOut from './helpers/test_out'
+import { Region } from "../src/types";
```
Comment (schickling) [inline tests/init.test.ts]: Remove semicolon

Comment (marktani) [inline tests/init.test.ts]: also, double quotes

**Pull request merged**